### PR TITLE
[tune] Directional metrics for components (#4120)

### LINKED ISSUE #4120: [tune] Trial scheduler should allow decreasing metric attribute
### Describe the problem
Right now, all the trial schedulers have a `reward_attr` attribute, which is assumed to be increasing. It would be nice if I could use decreasing metric as the scheduling metric.

### PR DESCRIPTION
**Pull request by hershg**
## What do these changes do?
Allows for a scheduler/searcher to either minimize or maximize the `metric` attribute (direction specified by `mode` parameter). Previously we only maximize over the attribute given by `reward_attr`.

## Related issue number

"Closes #4120"

## Linter

- [x] I've run `scripts/format.sh` to lint the changes in this PR.

### CODE REVIEW

**Comment by ericl** [comment]
It would be nice to not break backwards compatibility here.

How about adding metric but rewriting reward attr to that by default (and logging a warning to use metric instead)?

**Comment by richardliaw** [comment]
> It would be nice to not break backwards compatibility here.

This PR doesn't break backwards compat anywhere right? We leave reward_attr for all the Tune components (or did we miss one?)

File: python/ray/tune/schedulers/async_hyperband.py [hunk 1]
```diff
@@ -40,7 +41,9 @@ class AsyncHyperBandScheduler(FIFOScheduler):
 
     def __init__(self,
                  time_attr="training_iteration",
-                 reward_attr="episode_reward_mean",
+                 reward_attr=None,
+                 metric="episode_reward_mean",
```
Comment (richardliaw) [inline python/ray/tune/schedulers/async_hyperband.py]: cc @ericl

File: python/ray/tune/examples/tune_cifar10_gluon.py [hunk 2]
```diff
@@ -192,7 +192,8 @@ def test():
     elif args.scheduler == "asynchyperband":
         sched = AsyncHyperBandScheduler(
             time_attr="training_iteration",
-            reward_attr="neg_mean_loss",
+            rmetric="mean_loss",
```
Comment (richardliaw) [inline python/ray/tune/examples/tune_cifar10_gluon.py]:
```suggestion
            metric="mean_loss",
```

**Comment by ericl** [comment]
I see, I was looking at the wrong file.

**Review by richardliaw — APPROVED** [review]

**Pull request merged**